### ISSUE TITLE
Name conflict error during uploading file

### ISSUE THREAD

**Issue by kmtbang** [OneDrive/onedrive-sdk-ios#46]
We're using the ios sdk to upload files. It works fine for a new file. but if we try to upload the same file again, the sdk will return a 409  error as follows: Error Domain=com.microsoft.onedrive.errors Code=409 "conflict" UserInfo={error= nameAlreadyExists : An item with the same name already exists under the parent, NSLocalizedDescription=conflict}

Here is the request we are using to upload the file to our app folder :

  [[[[[[self.client drive] special:@"approot"] itemByPath:name] contentRequest] nameConflict:[ODNameConflict replace] ]  uploadFromFile:
        [NSURL fileURLWithPath:path] 
        completion:^(ODItem *response, NSError *error) {
            if (error == nil) {
            } else {
                NSLog(@"one drive upload:%@", error);
            }
        }
     ];

**Comment by aclev** [comment]
Hi @kmtbang What happens when you try it without the name conflict?

**Comment by kmtbang** [comment]
The same error happened without specifying name conflict. We added the name conflict part after seeing the error,  hoping it will resolve the issue. but the same error occurs.

**Comment by aclev** [comment]
It looks like this is a bug in our App Folders API.  To unblock you for now, is it possible you store the id of the item and use that to upload for future requests?

**Comment by kmtbang** [comment]
we'll try that.

By the way, when we tried the same upload code to upload the file to  the app folder in a One Drive for Business account, the uploading of new files always returns error of "unsupported url" as follows:

:Error Domain=NSURLErrorDomain Code=-1002 "unsupported url" UserInfo={NSUnderlyingError=0x7fa83cc05770 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}, NSErrorFailingURLStringKey=_api/v2.0/me/drive/special/approot:/Memo_20151013_123509.m4a:/content?@name.conflictBehavior=replace, NSErrorFailingURLKey=_api/v2.0/me/drive/special/approot:/Memo_20151013_123509.m4a:/content?@name.conflictBehavior=replace, NSLocalizedDescription=unsupported url}

The authorization with the One Drive Business account completed without any error. We used "myfiles.readwrite" and "myfiles.appfolder" for scopes during the authorization.

**Comment by aclev** [comment]
Currently AppFolder isn't supported in OneDrive for business, it's on our radar and we are working on it.  I don't have a specific timeline for it yet.

**Comment by kmtbang** [comment]
Hi,
   then for OneDrive for Business account , which folder can a third-party
app upload files to please?

2015-10-13 12:50 GMT-07:00 Ace Levenberg notifications@github.com:

**Comment by aclev** [comment]
if you have read/write access you can upload to any of the users folders.

**Comment by kmtbang** [comment]
Hi,
   we tried to upload to "Documents" folder, but always get "unsupported
url" error. When we authenticate the app with an OneDrive for Business
account, we use the scopes: "MyFiles.readwrite". The auth completed without
any error. Is the scope the correct one to use?

 We tried two ways to upload to the "Documents" folder as follows, but got
the same "unsupported url" error.
1.  [[[[client drive] special:@"documents"] itemByPath:filename]
   contentRequest] uploadFromFile:path completion:comletion
2.  [[[[client root] itemByPath"@"Documents"] itemByPath:filename]
   contentRequest] uploadFromFile:path completion:comletion
   
   Thanks.

2015-10-13 16:29 GMT-07:00 Ace Levenberg notifications@github.com:

**Comment by rgregg** [comment]
OneDrive for Business doesn't have a "documents" folder. Most users just treat the root of their OneDrive for Business as their documents folder, so you can just upload files directly there, or as a better alternative, create a new folder specific to the kinds of files you're uploading and upload them there.

**Comment by Ram-iOS** [comment]
Error getting while uploading : Error Domain=NSURLErrorDomain Code=-1005 "The network connection was lost." UserInfo={NSErrorFailingURLStringKey=https://api.onedrive.com/v1.0/drive/special/approot:/iSting_27_May_2016_17_05_23%0APart-2.Mov:/content, _kCFStreamErrorCodeKey=-4, NSErrorFailingURLKey=https://api.onedrive.com/v1.0/drive/special/approot:/iSting_27_May_2016_17_05_23%0APart-2.Mov:/content, NSLocalizedDescription=The network connection was lost., _kCFStreamErrorDomainKey=4, NSUnderlyingError=0x15ef14a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1005 "(null)" UserInfo={_kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-4}}}

**Comment by Ram-iOS** [comment]
if m try to upload video by interval than m getting this error, and if m upload video without interval it will upload successfully , 

please help me

**Comment by Ram-iOS** [comment]
The network connection was lost., _kCFStreamErrorDomainKey=4 , when try to upload video using interval

**Comment by kevklam** [comment]
Please create a new issue if you have a problem. This issue has been closed.
